v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -29,392 +29,585 @@ Note: More types will be added as needed.
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_005: [** If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall fail and return a non-zero value. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_006: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` shall fail and return a non-zero value. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_006: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall fail and return a non-zero value. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_007: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` shall copy the bytes of the `int64_t` value from the address pointed by `src_value` to `dst_value`. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_007: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall copy the bytes of the `int64_t` value from the address pointed by `src_value` to `dst_value`. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_008: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` shall succeed and return 0. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_008: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall succeed and return 0. **]**
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).free
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).free
 
 ```c
 typedef int (*LOG_CONTEXT_PROPERTY_TYPE_FREE)(void* value);
 ```
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_009: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).free` shall return. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_009: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).free` shall return. **]**
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).get_type
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).get_type
 
 ```c
 typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
 ```
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_010: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).get_type` shall returns the property type `LOG_CONTEXT_PROPERTY_TYPE_int64_t`. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_010: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).get_type` shall return the property type `LOG_CONTEXT_PROPERTY_TYPE_int64_t`. **]**
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string
+## LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)
+
+```c
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)(void* dst_value, int64_t src_value);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)` copies the `src_value` value to the address pointed by `dst_value`.
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_081: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_082: [** `LOG_CONTEXT_PROPERTY_TYPE_INIT(int64_t)` shall copy the `src_value` bytes of the `int64_t` to `dst_value`. **]**
```

Comment:
>shall copy the `src_value` bytes [](http://example.com/codeflow?start=93&length=32)

we should be mindful of alignment here, and if this is a way of saying "memcpy"... why?

`src_value` is already aligned because it comes as parameter on the stack.

what's the case for `dst_value` to be unaligned?

Why is `*(int64_t*)dst_value = src_value` avoided here, that's the question.

Reply:
Because when inheriting things from a context, we copy the arguments (at least at the moment). Maybe future iterations will make it smarter.
And since we copy the data, we do not really make sure to have the bytes aligned.
So for now I have to make this copied with memcpy. Let's see how it goes and we can always make this better with aligning it and assigning.